src/Log2.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+/// @notice Gas-efficient library for log2 calculations
+library Log2 {
```

Comment:
nit: I think this belongs next to the BitOps library in bitops.sol

Reply:
Since we've already built and vetted other bit ops stuff it might reduce our code surface and make this simpler if you just use what we've already built and do

```solidity
return 255 - clz(value) 
```

Does that sound ok?

Reply:
Didn't realise this can be used for log2 as well. Will dig into it and replace.

Reply:
Done.

Reply:
Yeah this was a cool tip. Thanks. I grok why it works.